components/form.js

```diff
@@ -0,0 +1,143 @@
+import React from "react";
+import Link from "next/link";
+
+export default class Form extends React.Component {
+  constructor() {
+    super();
+    this.state = { token: "", formFields: [], formData: {} };
+  }
+
+  // Make state targets equal to the value of input fields
+  change = e => {
+    this.setState({
+      formData: {
+        ...this.state.formData,
+        [e.target.name]: e.target.value
+      }
+    });
+  };
+
+  componentDidMount() {
+    // Fetch the data form fields from API, use it in state
+    let formFields = JSON.parse(localStorage.getItem("Form Fields"));
+    if (!(Array.isArray(formFields) && formFields.length)) {
+      fetch(
+        "https://rominacsvsms.api.ushahidi.io/api/v3/forms/14/attributes?order=asc&orderby=priority"
+      )
+        .then(response => {
+          if (response.ok) {
+            return response.json();
+          } else {
+            throw new Error("Field fetching response is not ok.");
```

Comment:
why tho? form fields are not necessarily in the system . You don't need to fail/throw an error for not having accessed the system before now and not having saved fields. 

Also what's the planned structure for this localStorage setup ?